### ISSUE TITLE
`ParamSpec`: mypy is happy for args to be annotated with `P.kwargs` and kwargs to be annotated with `P.args`

### ISSUE THREAD

**Issue by AlexWaygood** [python/mypy#13460]
**Bug Report**

Mypy has no issues with this snippet of code. It probably should:

```python
from typing import Callable
from typing_extensions import ParamSpec

P = ParamSpec("P")

def func(callback: Callable[P, str]) -> Callable[P, str]:
    def inner(*args: P.kwargs, **kwargs: P.args) -> str:
        return 'baz'
    return inner
```

This bug was accidentally discovered in #13459.

**To Reproduce**

https://mypy-play.net/?mypy=latest&python=3.10&gist=87f5f5277409f0fd77675e8dcc6a11fc

**Expected Behavior**

Mypy should emit two errors for this snippet:

1. `*args` have been annotated with `P.kwargs` instead of `P.args`
2. `**kwargs` have been annotated with `P.args` instead of `P.kwargs`

**Actual Behavior**

No error is emitted.
